tests/test_pickle.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from examples.seismic import demo_model
+from examples.seismic.source import TimeAxis, RickerSource
+from devito.dimension import TimeDimension, SteppingDimension
+from devito.base import Constant
+
+import cloudpickle as pickle
+
+
+def test_pickle():
```

Comment:
this needs to be split into multiple smaller tests

docstrings should be added